src/effects/VST3/VST3Utils.cpp

```diff
@@ -81,10 +81,8 @@ bool VST3Utils::ParseAutomationParameterKey(const wxString& key, Steinberg::Vst:
 
 Steinberg::IPtr<PresetsBufferStream> PresetsBufferStream::fromString(const wxString& str)
 {
-   Steinberg::Buffer buffer(str.size() / 4 * 3);
-   auto len = Base64::Decode(str, buffer);
-   assert(len <= buffer.getSize());
-   buffer.setSize(len);
+   auto data = str.To8BitData();
+   Steinberg::Buffer buffer(data, data.length());
```

Comment:
This commit is not strictly needed for statelessness.

You must have found a case where this appeared to fix persistency of settings in the configuration file.

The analogous code for VST2, which you originally adapted to VST3, still uses Base64 encoding and to change that would have some small risk of breaking compatibility of someone's settings configuration or macros if they migrate to 3.2.

What was the original reason Base64 encoding was used at all?  I'm not certain but I think it was a way of converting binary data into something that could be stored in a config file with proper escapes and no loss of data.  Is that right, and is it a reason why you still might want to figure out some other way to keep Base64 encoding working?

Reply:
>The analogous code for VST2, which you originally adapted to VST3, still uses Base64 encoding and to change that would have some small risk of breaking compatibility of someone's settings configuration or macros if they migrate to 3.2.

I haven't yet figured out what exactly causes data corruption. May be it's in Base64::Encode or Base64::Decode or may be there are some implicit charset conversion which wasn't taken into account.

>What was the original reason Base64 encoding was used at all? I'm not certain but I think it was a way of converting binary data into something that could be stored in a config file with proper escapes and no loss of data.

Correct